torch_geometric/nn/models/graphmask_explainer.py

```diff
@@ -0,0 +1,838 @@
+        prediction made by the GNN for graph :attr:`graph_idx`.
+
+        Args:
+            graph_idx (int): Graph to explain.
+
+        :rtype: (:class:`Tensor`)
+        """
+
+        if not isinstance(graph_idx, int):
+            raise ValueError(
+                "'graph_idx' parameter must be of type 'integer'.")
+
+        self.freeze_model(self.model)
+        self.__set_flags__(self.model)
+        self.model.eval()
+
+        with torch.no_grad():
+            latest_source_embeddings, latest_messages = [], []
+            latest_target_embeddings = []
+            for module in self.model.modules():
+                if isinstance(module, MessagePassing):
+                    latest_source_embeddings.append(
+                        module.latest_source_embeddings)
+                    latest_messages.append(module.latest_messages)
+                    latest_target_embeddings.append(
+                        module.latest_target_embeddings)
+            gate_input = [
+                latest_source_embeddings, latest_messages,
+                latest_target_embeddings
+            ]
+            if self.log:  # pragma: no cover
+                pbar = tqdm(total=self.num_layers)
+                pbar.set_description(f'Explain graph {graph_idx}')
+            for i in range(self.num_layers):
+                output = self.full_biases[i]
+                for j in range(len(gate_input)):
+                    partial = self.gates[i * 4][j](gate_input[j][i])
+                    result = self.gates[(i * 4) + 1][j](partial)
+                    output = output + result
+                relu_output = self.gates[(i * 4) + 2](output / len(gate_input))
+                sampling_weights = self.gates[(i * 4) +
+                                              3](relu_output).squeeze(dim=-1)
+                sampling_weights, _ = self.hard_concrete(
+                    sampling_weights, training=False)
+                if i == 0:
+                    edge_weight = sampling_weights
+                else:
+                    if (edge_weight.size(-1) != sampling_weights.size(-1)
+                            and self.layer_type == 'GAT'):
+                        sampling_weights = F.pad(
+                            input=sampling_weights,
+                            pad=(0, edge_weight.size(-1) -
+                                 sampling_weights.size(-1), 0, 0),
+                            mode='constant', value=0)
+                    edge_weight = torch.cat((edge_weight, sampling_weights), 0)
+                if self.log:  # pragma: no cover
+                    pbar.update(1)
+        if self.log:  # pragma: no cover
+            pbar.close()
+
+        edge_mask = edge_weight.view(-1,
+                                     edge_weight.size(0) // self.num_layers)
+        edge_mask = torch.mean(edge_mask, 0)
+
+        return edge_mask
+
+    def visualize_subgraph(self, node_idx: Optional[int], edge_index: Tensor,
```

Comment:
Any chance we can re-use functionality here from the base class?

Reply:
Actually, `visualize_subgraph()` of the base `Explainer` class [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/explainer.py) uses `threshold` parameter, whereas in this `GraphMaskExplainer` implementation, computation of threshold is being done on Line 767 with different formula and have also made minimal changes to the original [visualize_subgraph()](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/explainer.py) to plot the explanations properly. Therefore, don't think we can re-use the base `Explainer` class's `visualize_subgraph()` utility here.